tools/oeapkman/oeapkman

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+ARCH=$(arch)
+
+# Check if targeting OPTEE.
+optee=0
+if [ "$1" == "--optee" ]; then
+    # Optee support requires QEMU user-space emulation.
+    if ! command -v qemu-aarch64-static > /dev/null; then
+        echo "install qemu-user-static package for optee support."
+        exit 1
+    fi
+    optee=1
+    ARCH='aarch64'
+    shift
+fi
+
+# Use a recent version of proot. With older versions, apk would
+# report messages like "2 errors", that are innocuous, but catch
+# the eye.
+PROOT_URL="https://gitlab.com/proot/proot/-/jobs/981080842/\
+artifacts/download?file_type=archive"
+
+# URL to fetch alpine root fs from About 2.5 MB download.
+ALPINE_ROOTFS_URL="https://dl-cdn.alpinelinux.org/alpine/v3.14/\
+releases/$ARCH/alpine-minirootfs-3.14.0-${ARCH}.tar.gz"
+
+# Name of the folder where root filesystem will be extracted to.
+ALPINE_FS="alpine-fs-${ARCH}"
+
+# Figure out the user's home folder. If the command is executed
+# as sudo, then we use $SUDO_USER's home folder and not the
+# root's home folder.
+if [ -n "$SUDO_USER" ]; then
+    HOME_DIR=/home/${SUDO_USER}
+else
+    HOME_DIR=~
+fi
+
+OEAPKMAN_DIR=${HOME_DIR}/.oeapkman
+CURRENT_DIR=$(pwd)
+ROOTFS_PATH="${OEAPKMAN_DIR}/${ALPINE_FS}"
+
+LOCK_FILE="/var/lock/oeapkman.lock"
```

Comment:
It this isn't run as sudo, then it would be preferable to pick a path that's specific to the user to avoid unnecessary contention.